ASM/c/blue_warp.c

```diff
@@ -42,6 +42,16 @@ int32_t DoorWarp1_PlayerInRange_Overwrite(z64_actor_t* actor, z64_game_t* game)
             }
             extended_savectx.collected_dungeon_rewards[boss_idx] = true;
         }
+        // set time of day each time the blue warp is taken to reduce nonrepeatable access
+        if (boss_idx == 5) { // Morpha
+            z64_file.skybox_time = z64_file.day_time = 0x4800; // CLOCK_TIME(6, 45)
+        } else {
+            z64_file.skybox_time = z64_file.day_time = 0x8000; // CLOCK_TIME(12, 00)
+        }
+        // reset heat timer after Volvagia
+        if (boss_idx == 4) {
```

Comment:
While I'm on the readability train, idea for a separate chore commit: Could add a few #defines for these boss indices.

Reply:
This boss index concept is a bit weird and only really exists to help calculate offsets into the extended save context and delayed item list. Maybe for clarity, this should be `if (game->scene_index == SCENE_VOLVAGIA_BOSS_ROOM)` where `SCENE_VOLVAGIA_BOSS_ROOM` is defined as `0x0015` somewhere.